### ISSUE TITLE
Enforce branch policies on the repository

### ISSUE THREAD

**Issue by toddysm** [oras-project/oras-go#458]
To improve the security of the ORAS project we need to enforce the branch policies for this repository. I propose that we enforce the policies as follows:

- Use the following rules for `main` and `release/*` branches:
    - Require PR before merging
      - [ ] Require 3 approvals
      - [ ] Dismiss stale PR approvals when new commits are pushed
      - [ ] Require review from Code Owners
      - [ ] Require status checks to pass before merging
      - [ ] Require conversation resolution before merging
      - [ ] Require signed commits
      - [ ] Do not allow bypass the above settings

Please add your comments and proposals for additional changes to this issue.

**Comment by TerryHowe** [comment]
If a code owner created the PR, is that one approval?

**Comment by toddysm** [comment]
That is correct @TerryHowe - only codeowners count. And no, this is in addition to the person who submitted the PR as far as I know. We can have a relaxed policy and ask for 2 codeowner approvals only.

**Comment by shizhMSFT** [comment]
Few comments:
- `release/*` does not apply to libraries. Besides, we have a special branch named `v1`.
- "Require 3 approvals" is not applicable to this repository since we only have 2 active code owners.
